aten/src/ATen/native/xnnpack/Add.cpp

```diff
@@ -0,0 +1,228 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Allocator.h>
+#include <ATen/native/xnnpack/Common.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace add {
+
+struct Context final {
+  Operator add_op;
+
+  static constexpr float kMin = -std::numeric_limits<float>::infinity();
+  static constexpr float kMax = std::numeric_limits<float>::infinity();
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32 adds.
+
+bool available(
+    const int64_t input1_channels,
+    const int64_t input2_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+         // XNNPACK
+  return xnnpack::internal::available() &&
+         // Input1 == Input2
+         (input1_channels == input2_channels) &&
+         // Input == Output
+         (input1_channels == output_channels) &&
+         // Output Min / Max
+         (output_min.isIntegral(true) || output_min.isFloatingPoint()) &&
+         (output_max.isIntegral(true) || output_max.isFloatingPoint()) &&
+         (output_max.to<float>() > output_min.to<float>());
```

Comment:
what's the point of this comparison? As far as I can see everywhere this function is called it's called with constexpr args that make this comparison true.

Reply:
I understand it's necessary for clamp, but I don't see how it's useful in Add

Reply:
https://github.com/pytorch/vision/blob/791c172a337d98012018f98ffde93b1020ba3ed5/torchvision/models/resnet.py#L71

Add and ReLU can be fused.  There's actually a usecase for that in ResNet.

Reply:
Removing Add.